BonusCalcApi/V1/UseCase/UpdateReportSentAtUseCase.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Threading.Tasks;
+using BonusCalcApi.V1.Gateways.Interfaces;
+using BonusCalcApi.V1.Infrastructure;
+using BonusCalcApi.V1.UseCase.Interfaces;
+
+namespace BonusCalcApi.V1.UseCase
+{
+    public class UpdateReportSentAtUseCase : IUpdateReportSentAtUseCase
+    {
+        private readonly ITimesheetGateway _timesheetGateway;
+        private readonly IDbSaver _dbSaver;
+        public UpdateReportSentAtUseCase(ITimesheetGateway timesheetGateway, IDbSaver dbSaver)
+        {
+            _timesheetGateway = timesheetGateway;
+            _dbSaver = dbSaver;
+        }
+        public async Task ExecuteAsync(string operativeId, string weekId)
+        {
+            var timesheet = await _timesheetGateway.GetOperativeTimesheetAsync(operativeId, weekId);
+
+            if (timesheet is null) ThrowHelper.ThrowNotFound($"Timesheet not found for operative: {operativeId} and week: {weekId}");
```

Comment:
Consistency - Is there any reason the 404 is generated here rather than in the controller, other than you want to keep the return type simple

Reply:
That's how the [`UpdateTimesheetUseCase`][1] handles it so was keeping it the same to be consistent. Happy to change both to return the null value instead of throwing an error if you feel that's better?

[1]: https://github.com/LBHackney-IT/bonuscalc-api/blob/master/BonusCalcApi/V1/UseCase/UpdateTimesheetUseCase.cs#L26

Reply:
It's OK as is because in the second scenario, we lose a bit of error fidelity (null could mean two things)